File: kid-problem-ui-test-selenium/KpUiTestNUnit/Models/ExamRun.cs
namespace KpUiTestNUnit.Models;

public class ExamRun
{
    public string? UID { get; set; }
    public required string ExamCategory { get; set; }
    public required string ExamTitle { get; set; }
    public required string AnsweredBy { get; set; }
    public required string StartTime { get; set; }
    public string? CompleteTime { get; set; }
    public string? TotalDuration { get; set; }
    public required int TotalCount { get; set; }
    public int? CorrectCount { get; set; }
    public int? GuessCount { get; set; }
    public int? GuessCorrectCount { get; set; }

    public required ExamRunDetails[] ExamDetails { get; set; }

}

File: kid-problem-ui-test-playwright/models/ExamRun.ts
export class ExamRun {
    Uid: string | null | undefined;
    ExamCategory: string | null | undefined;
    ExamTitle: string | null | undefined;
    AnsweredBy: string | null | undefined;
    StartTime: string | null | undefined;
    CompleteTime: string | null | undefined;
    TotalDuration: string | null | undefined;
    TotalCount: string | null | undefined;
    CorrectCount: string | null | undefined;
    GuessCount: string | null | undefined;
    GuessCorrectCount: string | null | undefined;
    ExamDetails: ExamRunDetail[] | null | undefined;
}

export class ExamRunDetail {
    ProblemTitle: string | null | undefined;
    UserAnswer: string | null | undefined;
    Correct: string | null | undefined;
    Guess: string | null | undefined;
    Duration: string | null | undefined;

    toString(): string {
        return `${this.ProblemTitle}${this.UserAnswer}${this.Correct}${this.Guess}${this.Duration}`;
    }
}